components/datetime/benches/datetime.rs

```diff
@@ -95,6 +95,81 @@ fn datetime_benches(c: &mut Criterion) {
         true,
     );
 
+    let ten_cases = [
+        "2001-09-08T18:46:40.000[u-ca=gregory]",
+        "2017-07-13T19:40:00.000[u-ca=gregory]",
+        "2020-09-13T05:26:40.000[u-ca=gregory]",
+        "2021-01-06T22:13:20.000[u-ca=gregory]",
+        "2021-05-02T17:00:00.000[u-ca=gregory]",
+        "2021-08-26T10:46:40.000[u-ca=gregory]",
+        "2021-11-20T03:33:20.000[u-ca=gregory]",
+        "2022-04-14T22:20:00.000[u-ca=gregory]",
+        "2022-08-08T16:06:40.000[u-ca=gregory]",
+        "2033-05-17T20:33:20.000[u-ca=gregory]",
+    ]
+    .map(|s| DateTime::try_from_str(s, Gregorian).unwrap());
+
+    #[inline]
+    fn construct_any_ymd_numeric() -> DateTimeFormatter<fieldsets::YMD> {
+        DateTimeFormatter::try_new(locale!("fr").into(), fieldsets::YMD::short()).unwrap()
+    }
+
+    #[inline]
+    fn construct_fixed_ymd_numeric() -> FixedCalendarDateTimeFormatter<Gregorian, fieldsets::YMD> {
+        FixedCalendarDateTimeFormatter::try_new(locale!("fr").into(), fieldsets::YMD::short())
+            .unwrap()
+    }
+
+    group.bench_function("any/construct_and_format/ymd_numeric/10_cases", |b| {
+        let mut buffer = String::with_capacity(1000);
```

Comment:
issue: We should include the allocation costs and use `black_box(....write_to_string())`. I don't expect most users to be using the write_to API, we can have a separate benchmark for that if we wish but the common case (especially as mediated by FFI!) is going to be `write_to_string()` I think.

Reply:
I don't think so, allocation cost is not stable. We're trying to measure costs that we control, and while most users will have to add the string allocation cost, we don't control that.

Reply:
@robertbastian Benchmarks can have multiple goals: one goal is to keep track of performance regressions. Another goal is to understand the overall performance characteristics of a codepath so that you can 

The motivation for these is determining whether the optimizations from #5849 are worth it: for that we need to know if they are a significant win in the common case (with allocation) and if so by how much. 
I do not think these benchmarks serve that motivation unless they include allocation.


If allocation is the bulk of the runtime of a function, Amdahl's law means we probably should not spend too much effort optimizing the function itself.

Reply:
No, it depends on the type of workload. If they're formatting many dates in a loop, yes, but that's not a typical workload[^1]. A typical workload where perf matters is a bunch of scattered dates being formatted throughout an application, where the date formatting performance adds up. 


Perhaps poeple who can benefit from amortizing allocations are also worth optimizing (further) for, but that's not readily apparent to me in this situation at least. Happy to be convinced otherwise.

(In ICU4X I think we could be more diligent about talking about actual usage patterns when discussing performance)

 [^1]: Maybe a spreadsheet engine, but that's a highly specialized use case and I don't wish to optimize for it until someone with that use case actually approaches us with concrete needs.

Reply:
This is, essentially, the discussion I have wanted to have ever since #5849 was opened:

What are the actual _end to end_ situations we care about performance for? Why?

"We care about performance everywhere" is not useful when assessing tradeoffs, so knowing motivating use cases, and agreeing that they are realistic, and expected _enough_ that optimizing for them beforehand is worth it (rather than waiting on a client that needs it), is quite helpful.